puppet/ha-puppet/screenshot.js

```diff
@@ -264,12 +303,47 @@ export class Browser {
             polling: 100,
           },
         );
+
+        // Wait for all images and Lovelace cards to load
+        await page.waitForFunction(
+          () => {
+            // Check if all images are loaded
+            const images = Array.from(document.querySelectorAll("img"));
+            const allImagesLoaded = images.every(img => img.complete);
+
+            // Check if Lovelace UI is loaded (if we're on a Lovelace page)
+            const haEl = document.querySelector("home-assistant");
+            const mainEl = haEl?.shadowRoot?.querySelector("home-assistant-main");
+            const panelResolver = mainEl?.shadowRoot?.querySelector("partial-panel-resolver");
+            const panel = panelResolver?.children[0];
+
+            // Check if it's a Lovelace panel
+            if (panel && panel.tagName === "HUI-ROOT") {
+              const lovelaceView = panel.shadowRoot?.querySelector("hui-view");
+              if (!lovelaceView) return false;
+
+              // Check if cards are still loading
+              const cards = lovelaceView.shadowRoot?.querySelectorAll("hui-card-element-editor, hui-card");
+              if (!cards || cards.length === 0) return false;
```

Comment:
You'll now never succeed loading if there are 0 cards, which is a valid use case.

Reply:
Also, cards are being added a couple at a time, you actually don't have a guarantee that all cards have been added (although 200ms should make this a rare occurrence to be a false positive)